pkg/controller/certificates/trigger/policies/gatherer.go

```diff
@@ -46,25 +48,29 @@ func (g *Gatherer) DataForCertificate(ctx context.Context, crt *cmapi.Certificat
 		return Input{}, err
 	}
 
-	// Attempt to fetch the CertificateRequest resource for the current 'status.revision'.
+	// Fetch the CertificateRequest resource for the current
+	// 'status.revision' if it exists; use revision = 1 in case the initial
+	// issuance failed.
 	var req *cmapi.CertificateRequest
+	revision := 1
```

Comment:
The reason I made this change is that we now want to use `DataForCertificate` when the Certificate is failing.

And when the certificate is failing right after being created, the status.revision stays empty, meaning that `DataForCertificate` will return a nil CertificateRequest.

And since we need to fetch the CertificateRequest even if the issuer controller hasn't been able to set status.revision to 1, I decided to add this default revision.

I'm not sure if that's the correct approach though :-(

Reply:
This comment is utterly outdated, please do not read it 😅 Back in November I wasn't clear on what the "current" CR was.